src/anonymous-credentials/schema.ts

```diff
@@ -428,35 +459,41 @@ export class CredentialSchema extends Versioned {
     }
   }
 
-  static essential(): SchemaType {
-    return {
+  static essential(withDefinitions = true): IJsonSchema {
+    const s = {
+      // Currently only assuming support for draft-07 but other might work as well
+      [META_SCHEMA_STR]: 'http://json-schema.org/draft-07/schema#',
       type: 'object',
       properties: {
         [CRED_VERSION_STR]: { type: 'string' },
-        [SCHEMA_STR]: { type: 'string' }
+        [SCHEMA_STR]: { type: 'string' },
+        [SUBJECT_STR]: {}
       }
     };
-  }
-
-  forCredential(): object {
-    return { $version: this.version, ...this.schema };
+    if (withDefinitions) {
```

Comment:
you shouldn't mutate the schema like this, anyway, this method will not be used by any credential creation we have in the sdk etc

Reply:
Why not? I am mutating an object that i created in the same function so whats wrong with that?

Reply:
it feels wrong to be modifying the schema, especially for what seems an obscure method just to add definitions. i feel like it should result in the same object whether its for credential or for json encoding etc

Reply:
> it feels wrong to be modifying the schema, especially for what seems an obscure method just to add definitions

I am not modifying any user-given object but modifying what I created. I don't see the obscurity in creating an object based on flags passed as function arg.

> i feel like it should result in the same object whether its for credential or for json encoding etc

I agree that we can keep the custom definitions present in all cases without having any impact (except a bit of increase in   size of schema but thats not the concern) and thats the default behavior. The flag is allowing you to not have extra data in your schema when you know that you don't need to, eg. all the fields need string encoding. It's about being precise..